server/src/LspServer.ts

```diff
@@ -209,40 +213,81 @@ export class LspServer {
+        }
+      } catch (e) {
+        this.onRpcServerFindFailed();
+      }
     } else {
       command.addParameter(dbtPort.toString());
       try {
         await this.startDbtRpc(command, dbtPort);
       } catch (e) {
-        await this.showStartDbtRpcError(e instanceof Error ? e.message : `Failed to start dbt-rpc. ${String(e)}`);
+        this.onRpcServerStartFailed(e instanceof Error ? e.message : `Failed to start dbt-rpc. ${String(e)}`);
       }
     }
   }
 
-  async showStartDbtRpcError(message: string): Promise<void> {
-    console.log(message);
-    const actions = { title: 'Retry', id: 'retry' };
-    const errorMessageResult = await this.connection.window.showErrorMessage(message, actions);
-    if (errorMessageResult?.id === 'retry') {
-      await this.prepareRpcServer();
+  onPythonNotFound(): void {
+    this.onRpcServerStartFailed(
+      'Python was not found in your working environment. dbt Wizard requires valid python installation. Please visit [https://www.python.org/downloads](https://www.python.org/downloads).',
+    );
+  }
+
+  onRpcServerFindFailed(): void {
+    this.onRpcServerStartFailed(
+      `Failed to find dbt-rpc. You can use 'python3 -m pip install dbt-bigquery dbt-rpc' command to install it. Check in Terminal that dbt-rpc works running 'dbt-rpc --version' command or [specify the Python environment](https://code.visualstudio.com/docs/python/environments#_manually-specify-an-interpreter) for VS Code that was used to install dbt (e.g. ~/dbt-env/bin/python3).`,
+    );
+  }
+
+  onRpcServerStartFailed(message: string): void {
+    this.progressReporter.sendFinish();
+    this.connection.window.showErrorMessage(message);
+  }
+
+  async suggestToInstallDbt(python: string, dbtProfileType: string): Promise<void> {
+    const actions = { title: 'Install', id: 'install' };
+    const errorMessageResult = await this.connection.window.showErrorMessage(
+      `dbt is not installed. Would you like to install dbt, dbt-rpc and ${dbtProfileType} adapter?`,
+      actions,
+    );
+
+    if (errorMessageResult?.id === 'install') {
+      console.log(`Trying to install dbt, dbt-rpc and ${dbtProfileType} adapter`);
+      const installResult = await DbtUtilitiesInstaller.installDbt(python, dbtProfileType);
+      if (installResult.isOk()) {
+        this.connection.window.showInformationMessage(installResult.value);
+        await this.prepareRpcServer();
+      } else {
+        this.onRpcServerStartFailed(installResult.error);
+      }
+    } else {
+      this.onRpcServerFindFailed();
     }
   }
 
   logStartupInfo(contextInfo: DbtProfileResult, initTime: number, initDbtRpcAttempt: number): void {
     this.sendTelemetry('log', {
-      dbtVersion: getStringVersion(this.featureFinder.version),
-      python: this.featureFinder.python ?? 'undefined',
+      dbtVersion: getStringVersion(this.featureFinder?.versionInfo?.installedVersion),
+      python: this.python ?? 'undefined',
```

Comment:
Looks like we have `this.python` field only just for log it here. If it possible can you try to get rid of this field in LspServer class?